static/src/javascripts/projects/commercial/modules/messenger/background.js

```diff
@@ -38,50 +43,99 @@ const setBackground = (specs: AdSpec, adSlot: Node): Promise<any> => {
-    const backgroundParent = document.createElement('div');
-    backgroundParent.className = 'creative__background-parent';
-    if (specStyles.backgroundColor) {
-        backgroundParent.style.backgroundColor = specStyles.backgroundColor;
-    }
-    backgroundParent.appendChild(background);
+    // check to see whether the parent div exists already, if so, jut alter the style
+    const backgroundParentClass = 'creative__background-parent';
+    const backgroundClass = 'creative__background';
+
+    const maybeBackgroundParent = adSlot.querySelector(
+        `.${backgroundParentClass}`
+    );
+    const maybeBackground = maybeBackgroundParent
+        ? maybeBackgroundParent.querySelector(`.${backgroundClass}`)
+        : null;
+    const backgroundAlreadyExists = !!(
+        maybeBackgroundParent && maybeBackground
+    );
+
+    const getBackground = (): Promise<Background> => {
+        if (
+            maybeBackground &&
+            maybeBackgroundParent &&
+            backgroundAlreadyExists
+        ) {
+            return Promise.resolve({
+                backgroundParent: maybeBackgroundParent,
+                background: maybeBackground,
+            });
+        }
+        // Wrap the background image in a DIV for positioning. Also, we give
+        // this DIV a background colour if it is provided. This is because
+        // if we set the background colour in the creative itself, the background
+        // image won't be visible (think z-indexed layers)
+        const backgroundParent = document.createElement('div');
+
+        // Create an element to hold the background image
+        const background = document.createElement('div');
+        backgroundParent.appendChild(background);
 
-    if (specs.scrollType === 'fixed') {
         return fastdom
-            .read(() => {
-                if (adSlot instanceof Element) {
-                    return adSlot.getBoundingClientRect();
+            .write(() => {
+                if (backgroundParent) {
+                    adSlot.insertBefore(backgroundParent, adSlot.firstChild);
                 }
             })
-            .then(rect =>
-                fastdom.write(() => {
-                    if (rect) {
-                        background.style.left = `${rect.left}px`;
-                        background.style.right = `${rect.right}px`;
-                        background.style.width = `${rect.width}px`;
+            .then(() => ({ backgroundParent, background }));
+    };
+
+    const updateStyles = (
+        backgroundParent: HTMLElement,
+        background: HTMLElement
+    ) => {
+        backgroundParent.className = backgroundParentClass;
+        background.className = `${backgroundClass} creative__background--${specs.scrollType}`;
+
+        Object.assign(background.style, specStyles);
+
+        if (specStyles.backgroundColor) {
+            backgroundParent.style.backgroundColor = specStyles.backgroundColor;
```

Comment:
Would it be possible to wrap these into `fastdom.write()` too?

Reply:
Yep, good catch. Not sure why that wasn't in there originally since it will cause a repaint and thus should always be in a `fastdom` call (as I understand it).